src/rpc/client.zig

```diff
@@ -35,12 +17,16 @@ pub const FetchRpcError = Allocator.Error || HttpPostFetcher.Error || rpc.respon
 pub fn fetchRpc(
     fetcher: *HttpPostFetcher,
     allocator: Allocator,
-    request: anytype,
+    comptime method: rpc.methods.Method.Tag,
+    // NOTE: `std.meta.FieldType` causes eval branch quota to be hit
+    request: @typeInfo(rpc.methods.Method).Union.fields[@intFromEnum(method)].type,
 ) FetchRpcError!Response(@TypeOf(request)) {
-    const request_json = try rpc.request.serialize(allocator, request);
+    const request_json = try std.json.stringifyAlloc(allocator, request, .{});
```

Comment:
This won't work. The tests aren't catching this bug because you changed the tests to run different code than the client.